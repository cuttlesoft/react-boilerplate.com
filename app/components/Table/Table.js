/* eslint-disable react/prop-types */
import React from 'react'
import { Table } from 'grommet'

import { doc } from './Table.doc'
import helpers from '../../utils/helpers'

/**
 *
 * Table
 *
 */
function createTable({ a11lyTitle, alignSelf, caption, children, margin, ...rest }) {
  return (
    <Table
      a11lyTitle={a11lyTitle}
      alignSelf={alignSelf}
      caption={caption}
      margin={margin}
      {...rest}
    >
      {children}
    </Table>
  )
}

export default helpers.createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: createTable,
})
