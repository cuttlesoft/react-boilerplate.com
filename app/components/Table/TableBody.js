/* eslint-disable react/prop-types */
import React from 'react'
import { TableBody } from 'grommet'

import { doc } from './TableBody.doc'
import helpers from '../../utils/helpers'

/**
 *
 * TableBody
 *
 */
function createTableBody({ children, ...props }) {
  return <TableBody {...props}>{children}</TableBody>
}

export default helpers.createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: createTableBody,
})
