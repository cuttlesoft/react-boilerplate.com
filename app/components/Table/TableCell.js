/* eslint-disable react/prop-types */
import React from 'react'
import { TableCell } from 'grommet'

import { doc } from './TableCell.doc'
import helpers from '../../utils/helpers'

/**
 *
 * TableCell
 *
 */
function createTableCell({ children, plain, scope, size, verticalAlign, ...props }) {
  return (
    <TableCell plain={plain} scope={scope} size={size} verticalAlign={verticalAlign} {...props}>
      {children}
    </TableCell>
  )
}

export default helpers.createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: createTableCell,
})
