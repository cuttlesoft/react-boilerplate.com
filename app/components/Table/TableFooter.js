/* eslint-disable react/prop-types */
/**
 *
 * TableFooter
 *
 */

import React from 'react'
import { TableFooter } from 'grommet'

function createTableFooter({ children, ...props }) {
  return <TableFooter {...props}>{children}</TableFooter>
}

let createTableFooterWithDoc

if (process.env.NODE_ENV !== 'production') {
  createTableFooterWithDoc = require('./TableFooter.doc').doc(createTableFooter) // eslint-disable-line global-require
}

const TableFooterWrapper = createTableFooterWithDoc || createTableFooter
export default TableFooterWrapper
