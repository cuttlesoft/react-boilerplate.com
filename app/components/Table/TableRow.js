/* eslint-disable react/prop-types */
import React from 'react'
import { TableRow } from 'grommet'

import { doc } from './TableRow.doc'
import helpers from '../../utils/helpers'

/**
 *
 * TableRow
 *
 */
function createTableRow({ children, ...props }) {
  return <TableRow {...props}>{children}</TableRow>
}

export default helpers.createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: createTableRow,
})
