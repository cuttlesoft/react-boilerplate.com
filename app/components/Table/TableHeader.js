/* eslint-disable react/prop-types */
/**
 *
 * TableHeader
 *
 */

import React from 'react'
import { TableHeader } from 'grommet'
import { doc } from './TableFooter.doc'
import helpers from '../../utils/helpers'

function createTableHeader({ children, ...props }) {
  return <TableHeader {...props}>{children}</TableHeader>
}

export default helpers.createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: createTableHeader,
})
