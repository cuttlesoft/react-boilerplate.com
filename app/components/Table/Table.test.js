/**
 *
 * Tests for Table
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Table from './Table'

const options = ['one', 'two', 'three']
const renderComponent = (props = {}) =>
  render(<Table label="Table" {...props} options={options} />)

test('it renders and matches snapshot', () => {
  const { container } = renderComponent()
  expect(container).toMatchSnapshot()
})
