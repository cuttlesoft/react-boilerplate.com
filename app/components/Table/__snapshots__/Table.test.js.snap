// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders and matches snapshot 1`] = `
<div>
  <table
    class="StyledTable-sc-1m3u5g-6 bJymoW"
  >
    <thead
      class="StyledTable__StyledTableHeader-sc-1m3u5g-4 kdPLsZ"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2 dSAyNg"
      >
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 gsRqPi"
          scope="col"
        >
          <h4
            class="StyledHeading-sc-1rdh4aw-0 cJdkwo"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          >
            Name
          </h4>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 gsRqPi"
          scope="col"
        >
          <h4
            class="StyledHeading-sc-1rdh4aw-0 cJdkwo"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          >
            Email
          </h4>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 gsRqPi"
          scope="col"
        >
          <h4
            class="StyledHeading-sc-1rdh4aw-0 cJdkwo"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          >
            Amount
          </h4>
        </th>
      </tr>
    </thead>
    <tbody
      class="StyledTable__StyledTableBody-sc-1m3u5g-3 bMdIOJ"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2 dSAyNg"
      >
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 glVxec"
          scope="col"
        >
          <h5
            class="StyledHeading-sc-1rdh4aw-0 hetXvW"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          >
            Eric
          </h5>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 glVxec"
          scope="col"
        >
          <h5
            class="StyledHeading-sc-1rdh4aw-0 hetXvW"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          >
            eric@gmail.com
          </h5>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 glVxec"
          scope="col"
        >
          <h5
            class="StyledHeading-sc-1rdh4aw-0 hetXvW"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          >
            $45
          </h5>
        </th>
      </tr>
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2 dSAyNg"
      >
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 glVxec"
          scope="col"
        >
          <h5
            class="StyledHeading-sc-1rdh4aw-0 hetXvW"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          >
            Chris
          </h5>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 glVxec"
          scope="col"
        >
          <h5
            class="StyledHeading-sc-1rdh4aw-0 hetXvW"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          >
            chris@gmail.com
          </h5>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 glVxec"
          scope="col"
        >
          <h5
            class="StyledHeading-sc-1rdh4aw-0 hetXvW"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          >
            $45
          </h5>
        </th>
      </tr>
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2 dSAyNg"
      >
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 glVxec"
          scope="col"
        >
          <h5
            class="StyledHeading-sc-1rdh4aw-0 hetXvW"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          >
            Shanna
          </h5>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 glVxec"
          scope="col"
        >
          <h5
            class="StyledHeading-sc-1rdh4aw-0 hetXvW"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          >
            shanna@gmail.com
          </h5>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 glVxec"
          scope="col"
        >
          <h5
            class="StyledHeading-sc-1rdh4aw-0 hetXvW"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          >
            $45
          </h5>
        </th>
      </tr>
    </tbody>
    <tfoot
      class="StyledTable__StyledTableFooter-sc-1m3u5g-5 iKFIhK"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2 dSAyNg"
      >
        <td
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 kgsbPu"
        >
          <h4
            class="StyledHeading-sc-1rdh4aw-0 cJdkwo"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          />
        </td>
        <td
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 kgsbPu"
        >
          <h4
            class="StyledHeading-sc-1rdh4aw-0 cJdkwo"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          />
        </td>
        <td
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 kgsbPu"
        >
          <h4
            class="StyledHeading-sc-1rdh4aw-0 cJdkwo"
            style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
          />
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;
